misc.go

```diff
@@ -55,6 +55,13 @@ func (t statusCodeError) HTTPStatusCode() int {
 	return t.Code
 }
 
+func (t statusCodeError) Retryable() bool {
+	if t.Code >= 500 || t.Code == http.StatusTooManyRequests {
```

Comment:
when will this method ever return false.... =)

Reply:
oops e09ebc0

Reply:
TBH I wouldn't expect a 500 to be something that can be retried. but i'm going to let it slide for now. I reserve the right to remove it later.